cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -81,4 +128,202 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+    }
+    if (folderName == null || folderName.length() == 0 || dataFileName == null || dataFileName.length() == 0) {
+      File[] pipelineFiles =
+        uuidFile.listFiles((dir, name) -> !name.startsWith(".") && !dir.isHidden() && dir.isDirectory());
+      List<SupportBundleFiles> supportBundleFilesList = new ArrayList<>();
+      if (pipelineFiles != null && pipelineFiles.length > 0) {
+        for (File pipelineFile : pipelineFiles) {
+          if (folderName != null && folderName.length() > 0) {
+            if (pipelineFile.getName().equals(folderName)) {
+              SupportBundleFiles supportBundleFiles = bundleService.getFilesName(pipelineFile);
+              responder.sendString(HttpResponseStatus.OK, GSON.toJson(supportBundleFiles));
+              return;
+            }
+          } else {
+            supportBundleFilesList.add(bundleService.getFilesName(pipelineFile));
+          }
+        }
+      }
+      responder.sendString(HttpResponseStatus.OK, GSON.toJson(supportBundleFilesList));
+      return;
+    }
+    File folderDirectory = new File(uuidFile, folderName);
+    File[] dataFiles =
+      folderDirectory.listFiles((dir, name) -> !name.startsWith(".") && !dir.isHidden() && dir.isDirectory());
+    if (dataFiles != null && dataFiles.length > 0) {
+      for (File dataFile : dataFiles) {
+        if (dataFile.getName().startsWith(dataFileName)) {
+          BufferedReader br = new BufferedReader(new FileReader(dataFile));
+          StringBuilder sb = new StringBuilder();
+          String line = br.readLine();
+          while (line != null) {
+            sb.append(line).append("\n");
+            line = br.readLine();
+          }
+          responder.sendString(HttpResponseStatus.OK, sb.toString());
+          return;
+        }
+      }
+    }
+    responder.sendString(HttpResponseStatus.NOT_FOUND,
+                         String.format("uuid: %s with pipeline file name: %s and data file name: %s not found", uuid,
+                                       folderName, dataFileName));
+  }
+
+  /**
+   * Exports all selected files as a ZIP archive file.
+   *
+   * @param uuid the bundle id which is also the uuid fileJson body
+   */
+  @POST
+  @Path("/support/bundle/{uuid}/download")
+  public void appsExport(FullHttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId supportBundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(supportBundleEntityId, StandardPermission.GET);
+    String requestContent = request.content().toString(StandardCharsets.UTF_8);
+    String applicationZip = "application/zip";
+    String contentDescriptionValue = "attachment; filename=\"collect_support_bundle.zip\"";
+    if (requestContent == null || requestContent.length() == 0) {
+      throw new BadRequestException("Request body is empty.");
+    }
+    JsonObject fileBody = GSON.fromJson(requestContent, JsonObject.class);
```

Comment:
We should have a `SupportBundleExportRequest` or something similar to encapsulate what the schema input for a call to this endpoint will look like instead of manually serializing/deserializing from a JSON object. Same for response.